test/list_test.dart

```diff
@@ -1066,4 +1066,82 @@ Future<void> main([List<String>? args]) async {
     expect(game.winnerByRound, isA<RealmList<Player>>());
     expect(game.winnerByRound, isA<RealmList<Player?>>());
   });
+
+  test('Move equality & hash', () {
+    expect(Move(1, 1), equals(Move(1, 1)));
+    expect(Move(1, 1), isNot(equals(Move(2, 2))));
+    expect(Move(1, 1).hashCode, equals(Move(1, 1).hashCode));
+    expect(Move(1, 1).hashCode, isNot(equals(Move(2, 2).hashCode)));
+  });
+
+  test('List.move', () {
+    final list = [0, 1, 2, 3];
+    list.move(1, 0);
+    expect(list, [1, 0, 2, 3]);
+    list.move(2, 3);
+    expect(list, [1, 0, 3, 2]);
+    final length = list.length;
+    expect(() => list.move(-1, 0), throwsRangeError);
+    expect(() => list.move(0, -1), throwsRangeError);
+    expect(() => list.move(length, 0), throwsRangeError);
+    expect(() => list.move(0, length), throwsRangeError);
```

Comment:
What about moving from one index to the same index. Can we add it to the tests? Also for managed realm list.

Reply:
That is a no-op, but sure I can add a test